src/service/tcp.rs

```diff
@@ -18,6 +18,7 @@ use crate::{
 };
 
 const INITIAL_TRANSACTION_ID: TransactionId = 0;
+const MAX_RETRIES: usize = 10;
```

Comment:
Max retries of what? Why 10? If this is an arbitrary number then it should better be configurable instead of a hard-coded constant.

A comment with references why this retry algorithm is needed is also required. Either by documenting the constant or the corresponding configuration option.

Reply:
Let **MAX_RETRIES** be configurable is more suitable. Can you give me some advice? I have not found how to add configurable code in this crate.

The retry algorithm is inspired by **libmodbus**'s error recovery mode, see [https://libmodbus.org/reference/modbus_set_error_recovery/](url) for details.

> When MODBUS_ERROR_RECOVERY_PROTOCOL is set, a sleep and flush sequence will be used to clean up the ongoing communication, this can occurs when the message length is invalid, the TID is wrong or the received function code is not the expected one. The response timeout delay will be used to sleep.

Reply:
If this parameter should be configurable then make a proposal. Otherwise use a constant and comment (in the code) why.